### ISSUE TITLE
Cannot work on newly created image without

### ISSUE THREAD

**Issue by vcarluer** [oliver-moran/jimp#147]
I cannot blit a newly created Jimp image to another newly create Jimp image.
The only way I made it work was saving the first image and reading it again to blit the second one in.

**Comment by oliver-moran** [comment]
Interesting. Can't think of a reason why not. Could you post a test case and I'll look into it.

**Comment by andyfenelon** [comment]
i'm having this issue too,  here's the code i'm trying

it's writing to disk but it seems to be resolving the promise 

> resolve("./edited.png")

before  and giving the error 

> Error: ENOENT: no such file or directory, open './edited.png'

```
const saveBuffer = (buffer) => {
  return new Promise((resolve, reject) => {
    var bitmap = new Buffer(buffer, 'base64');
    fs.writeFile('img.png', bitmap, (err) => {
      if (err) throw err;
      console.log('It\'s saved!');
      resolve('img.png')
    });
  })
}

const editImg = (img) => {
  return new Promise((resolve, reject) => {
    console.log(img)
    Jimp.read('./img.png')
      .then(function (image) {
        image.resize(256, 256)
         .quality(60)
         //.gaussian( 10 )
         .greyscale()
         .write("edited.png");
       })
       .then(res => {
         //console.log(res)
         resolve("./edited.png")
       })
       .catch(function (err) {
         console.log(err)
         throw err
       });
  })
}

const base64Encode = (file) => {
  return new Promise((resolve, reject) => {
    var bitmap = fs.readFileSync(file);
    console.log('base64Encode')
    resolve(new Buffer(bitmap).toString('base64'));
  })
}

app.post('/upload', (req, res) => {
  co(function* () {
    var img = yield saveBuffer(req.body);
    var edited = yield editImg(img);
    var base64 = yield base64Encode(edited);
    return base64;
  }).then(function (value) {
    //console.log(value)
    res.json({
      img: value
    })
  }, function (err) {
    console.error(err.stack);
  });

})
```

**Comment by vcarluer** [comment]
In fact I think I made it work. 
I have 2 scripts and I manage to make it work on the simpler one.
I don't have time to retry with the more complex one but when reading it I think it is because I used the return var from the new Jimp() constructor and not the one in the callback.
This was a recursive function which was creating poker cards on an A4 background so the code should be refactored to work with the callback instead of the returned var.

In your doc:
var image = new Jimp(256, 256, 0xFF0000FF, function (err, image) {
    // this image is 256 x 256, every pixel is set to 0xFF0000FF 
});

I used somewhere the 'var image' which was not ready at the moment and I try to blit from it.
I also have a case (I don't know why I did that...) I used the return var in the callback and another one which is a global page Jimp image var which is certainly overwritten when Jimp still trying to write to it.
Bad usage from my part so.

var card = new Jimp(256, 256, 0xFF0000FF, function (err, image) {
    // page is global and defined before (and recreated for each page)
    page.blit(card, x, y)
});

I still have one question: What is the interest of the returned jimp var if it cannot be used? In javascript the new will always return a var but maybe remove it from your examples?

Thank you for your reply anyway.
